lib/origen/parameters.rb

```diff
@@ -100,10 +100,31 @@ def _live_parameter_requested?
 
     private
 
-    def _validate_parameter_set_name(name)
-      unless _parameter_sets.key?(name)
-        puts "Unknown parameter set :#{name} requested for #{self.class}, these are the valid sets:"
-        _parameter_sets.keys.each { |k| puts "  :#{k}" }
+    def _validate_parameter_set_name(expr)
```

Comment:
Don't need to change it, but just FYI, since this method is private there is not really any need to make it begin with an underscore.
Normally I would make a method start with an underscore if it was really intended for internal use only but for some reason it could not be made private. I would also mark it as private in the documentation as shown below:

~~~ruby

# @api private
def _an_internal_use_method_but_which_is_not_private
end

private

def a_private_method
end
~~~

Reply:
:-) just realized that I may have actually been advising my past self there, but I'll leave it here as a useful note to whoever may see it.